Firestore/core/src/firebase/firestore/core/sync_engine.cc

```diff
@@ -281,13 +281,19 @@ void SyncEngine::HandleRejectedListen(TargetId target_id, Status error) {
     // store's invariants with another method.
     NoDocument doc(limbo_key, SnapshotVersion::None(),
                    /* has_committed_mutations= */ false);
-    DocumentKeySet limbo_documents = DocumentKeySet{limbo_key};
-    RemoteEvent event{SnapshotVersion::None(), /*target_changes=*/{},
-                      /*target_mismatches=*/{},
-                      /*document_updates=*/{{limbo_key, doc}},
+
+    // Explicitly instantiate these to work around a spurious "chosen
```

Comment:
Note: what I think happens is that GCC implemented both the default constructor and the constructor taking a single integer argument (bucket count) as the same constructor with a default argument value. This is incorrect because one of those has to be `explicit` (the one taking an integer argument) and the other one has to be _not_ `explicit` (the default constructor). This implementation, however, has the effect of making both of them, default constructor included, `explicit`. An `explicit` default constructor prevents this form of initialization:
```
foo = {}; // Error if default constructor of `Foo` is explicit
foo = Foo{}; // Ok
```

It's just for the record -- feel free to incorporate any details you find useful into the comment, or just leave as is.

Reply:
Good points, all. PTAL.